src/dwarf.cpp

```diff
@@ -1109,6 +1109,57 @@ attribute_value dwarf::implementation::process_form(const attribute& attr,
         result.reference(static_cast<std::uint32_t>(cu_offset + offset));
     };
 
+    const auto handle_passover = [&](){
+        // We have a problem if we are passing over an attribute that is needed to determine
+        // ODRVs.
+        assert(nonfatal_attribute(attr._name));
+        result.passover();
+        auto size = form_length(attr._form, _s);
+        _s.seekg(size, std::ios::cur);
+    };
+
+    enum class block_type {
+        one,
+        two,
+        four,
+        uleb,
+    };
+
+    const auto handle_block = [&](std::size_t size) {
```

Comment:
Where the handling of an essential block takes place. We get a `size` amount from `maybe_handle_block` telling us how many bytes are in this block that we need to process. We read them one at a time, accumulating them in an unsigned 64-bit value. This assumes the value is both an integer, and will fit in 64 bits. If either of this is found to be false, we'll need to revisit this.